### PR TITLE
CSPL-1027: Utility to download or upload file to s3

### PR DESCRIPTION
**Pull request by tpereirasplunk**
* Created utility to create a s3 session

* Updated Utility to be able to download from S3
** Utility can now be used to download any generic file

* Created Utility to be able to upload file to S3

* Created Utility to delete file on S3

### CODE REVIEW

File: test/secret/secret_m4_test.go [hunk 1]
```diff
@@ -43,7 +43,7 @@ var _ = Describe("Secret Test for M4 SVA", func() {
 		}
 	})
 
-	Context("Multisite cluster deployment (M13 - Multisite indexer cluster, Search head cluster)", func() {
+	XContext("Multisite cluster deployment (M13 - Multisite indexer cluster, Search head cluster)", func() {
```
Comment (smohan-splunk) [inline test/secret/secret_m4_test.go:46]: Why are we disabling this?

File: test/secret/secret_s1_test.go [hunk 2]
```diff
@@ -123,7 +123,7 @@ var _ = Describe("Secret Test for SVA S1", func() {
 		})
 	})
 
-	Context("Standalone deployment (S1) with LM", func() {
+	XContext("Standalone deployment (S1) with LM", func() {
```
Comment (smohan-splunk) [inline test/secret/secret_s1_test.go:126]: Why are we disabling this?

File: test/secret/secret_m4_test.go (hunk 1 above)
Comment (tpereirasplunk) [reply test/secret/secret_m4_test.go:46]: Accidental test code....... Reverting it.

File: test/secret/secret_s1_test.go (hunk 2 above)
Comment (tpereirasplunk) [reply test/secret/secret_s1_test.go:126]: Accidental test code....... Reverting it.

File: test/testenv/s3utils.go [hunk 3]
```diff
@@ -64,27 +64,58 @@ func CheckPrefixExistsOnS3(prefix string) bool {
...
+		logf.Log.Error(err, "Failed to download file")
 	}
 
 	logf.Log.Info("Downloaded", "filename", file.Name(), "bytes", numBytes)
 	return file.Name(), err
 }
+
+// UploadFileToS3 upload file to S3
+func UploadFileToS3(filename string, filepath string, file *os.File) (string, error) {
+	dataBucket := testS3Bucket
```
Comment (smohan-splunk) [inline test/testenv/s3utils.go:106]: Do we need separate environment variables for Apps? Currently the same (TEST_BUCKET) seems to be used for both license file and apps

File: test/testenv/s3utils.go [hunk 4]
```diff
@@ -64,27 +64,58 @@ func CheckPrefixExistsOnS3(prefix string) bool {
...
 	location := enterpriseLicenseLocation
 	item := "enterprise.lic"
-	file, err := os.Create(item)
+	filename, err := DownloadFileFromS3(item, location)
+	return filename, err
+}
+
+// S3Session Create session object for S3 bucket connection
+func S3Session() *session.Session {
+	sess, _ := session.NewSession(&aws.Config{Region: aws.String(s3Region)})
```
Comment (jryb) [inline test/testenv/s3utils.go:77]: If this errors out, is there no reason why we would need the error string?

**Comment by pdhanoyasplunk** [comment]
`Created utility to create a s3 session ` I am unable to find a method for this ?

File: test/testenv/s3utils.go (hunk 3 above)
Comment (pdhanoyasplunk) [inline test/testenv/s3utils.go:106]: Would be a good idea to accept this as a parameter. Make the method more modular.

File: test/testenv/s3utils.go (hunk 4 above)
Comment (pdhanoyasplunk) [reply test/testenv/s3utils.go:77]: We should handle the error here and in callers as well.

**Review by smohan-splunk — APPROVED** [review]

File: test/testenv/s3utils.go (hunk 3 above)
Comment (tpereirasplunk) [reply test/testenv/s3utils.go]: Modified

Comment (tpereirasplunk) [reply test/testenv/s3utils.go]: Modified

File: test/testenv/s3utils.go (hunk 4 above)
Comment (tpereirasplunk) [reply test/testenv/s3utils.go]: Modified

**Comment by jryb** [comment]
Could you squash all these commits into one? It should cancel out to just your changes.  Other than that looks good.

**Review by jryb — APPROVED** [review]

**Review by kashok-splunk — APPROVED** [review]

**Review by jambrosiano — APPROVED** [review]

**Review by pdhanoyasplunk — APPROVED** [review]

**Pull request merged**